compiler/surface/name_resolution.ml

```diff
@@ -563,9 +669,18 @@ let get_def_key
               ScopeVar.format_t x_uid
           else None )
   | [y; x] ->
-    let subscope_uid : SubScopeName.t = get_subscope_uid scope_uid ctxt y in
-    let subscope_real_uid : ScopeName.t =
-      SubScopeMap.find subscope_uid scope_ctxt.sub_scopes
+    let (subscope_uid, subscope_real_uid) : SubScopeName.t * ScopeName.t =
+      match
+        Desugared.Ast.IdentMap.find_opt (Marked.unmark y) scope_ctxt.var_idmap
+      with
+      | Some (SubScope (v, u)) -> v, u
+      | Some _ ->
+        Errors.raise_spanned_error pos
+          "Invalid access to input variable, '%s' is not a subscope"
```

Comment:
Ok so in some error messages I had stylised the names and identifiers coming from the source code in yellow, could be worth doing the same here. I maintain we should hire an intern to port https://docs.rs/ariadne/0.1.3/ariadne/ to OCaml and integrate it to the Catala compiler.